### ISSUE TITLE
[DEV] xdcp from non-root user copies files as root by default

### ISSUE THREAD

**Issue by pdlun92** [xcat2/xcat-core#1001]
In the following example, `vhu` is xCAT enabled non-root user.

```
[vhu@fs2vm100 ~]$ xdcp fs2vm101 /tmp/victor_file /tmp/testcopy/
[vhu@fs2vm100 ~]$ xdsh fs2vm101 ls -l /tmp/testcopy
fs2vm101: total 0
fs2vm101: -rw-r--r-- 1 root root 0 Apr 28 11:46 victor_file
```

Work around is to use `-l` option to specify user

```
[vhu@fs2vm100 ~]$ xdcp fs2vm101 -l vhu /tmp/victor_file /tmp/testcopy/
[vhu@fs2vm100 ~]$ xdsh fs2vm101 ls -l /tmp/testcopy
fs2vm101: total 0
fs2vm101: -rw-r--r-- 1 vhu users 0 Apr 28 11:48 victor_file
```

@daniceexi  should xdcp issued from non-root user preserve user by default?

**Comment by daniceexi** [comment]
@pdlun92 could you take a look whether the **xdsh -p|--preserve** option works for you?

**Comment by pdlun92** [comment]
@daniceexi seems to be the same
```
[vhu@fs2vm100 ~]$ xdcp fs2vm101 -p /tmp/victor_file /tmp/testcopy
[vhu@fs2vm100 ~]$ xdsh fs2vm101 ls -l /tmp/testcopy
fs2vm101: total 64
fs2vm101: -rw-rw-r-- 1 root root 4 May 10 15:31 victor_file
```

**Comment by whowutwut** [comment]
@neo954   I think we had talked about this on the interlock call.   The issue here is that a non root user has been given permission to use xdcp.  (distributed copy).  As a non root user, using `xdcp` to copy a file over to another node results in that file being created as the root user.  This is the bug. 

If we look at normal scp, the behavior preserves the ownership of the file.  

Taking two VMs with the `vhu` user ...

1) Create a file: 
```
[vhu@c910f04x42vm101 ~]$ touch vm101_testfile
```
2) Send the file over to vm106 as the user... 
```
[vhu@c910f04x42vm101 ~]$ scp vm101_testfile vhu@c910f04x42vm106:/tmp
Warning: Permanently added 'c910f04x42vm106,10.4.42.106' (RSA) to the list of known hosts.
vhu@c910f04x42vm106's password:
vm101_testfile                                100%    0     0.0KB/s   00:00
```
3) Verify the file on vm106... 
```
[vhu@c910f04x42vm101 ~]$ ssh vhu@c910f04x42vm106
vhu@c910f04x42vm106's password:
[vhu@c910f04x42vm106 ~]$ ls -ltr /tmp/vm101_testfile
-rw-rw-r-- 1 vhu vhu 0 Aug 16 14:08 /tmp/vm101_testfile
```
 
So this is an issue with xCAT that we do not preserve the owner of the file when doing xdcp.

**Comment by whowutwut** [comment]
@gurevichmark  do you have any cycles to look at this in the next few weeks?

**Comment by gurevichmark** [comment]
Yes. Starting to work on the solution.

**Comment by gurevichmark** [comment]
Fixed by pull request #2689. 
Solution is to follow logic used by `xdsh -K`

**Comment by gurevichmark** [comment]
Closing for @pdlun92 who is not part of xCAT team currently.
